coltypes/coltypes.go

```diff
@@ -127,103 +130,61 @@ func GetColumnTypes(ctx context.Context, repo *repository.Oracle, tableName stri
 	return columnTypes, nil
 }
 
-// ConvertStructureData converts a sdk.StructureData values to a proper database types.
-func ConvertStructureData(
+// FormatData formats the data into Oracle types
+// and wraps the placeholder in a special function, if necessary.
+func FormatData(
 	columnTypes map[string]ColumnData,
-	data sdk.StructuredData,
-) (sdk.StructuredData, error) {
-	result := make(sdk.StructuredData, len(data))
-
-	for key, value := range data {
-		if value == nil {
-			result[key] = nil
+	key string, value any,
+) (string, any, error) {
+	if value == nil {
+		return QueryPlaceholder, nil, nil
+	}
 
-			continue
+	switch oracleColumnData := columnTypes[strings.ToUpper(key)]; {
+	// if the column type is a NUMBER, precision is 1, scale is 0, and value is a boolean,
+	// convert it to the integer, when the "true" value is 1, and "false" is 0
+	case oracleColumnData.Type == oracleTypeNumber &&
+		oracleColumnData.Precision != nil && *oracleColumnData.Precision == 1 &&
+		oracleColumnData.Scale != nil && *oracleColumnData.Scale == 0:
+		valueBool, ok := value.(bool)
+		if !ok {
+			return QueryPlaceholder, value, nil
 		}
 
-		switch reflect.TypeOf(value).Kind() {
-		case reflect.Map, reflect.Slice: // convert the map type to the string, for the VARCHAR2 Oracle type
-			bs, err := json.Marshal(value)
-			if err != nil {
-				return nil, fmt.Errorf("marshal map or slice: %w", err)
-			}
-
-			result[key] = string(bs)
-
-			continue
+		if valueBool {
+			return QueryPlaceholder, 1, nil
 		}
 
-		switch oracleColumnData := columnTypes[strings.ToUpper(key)]; {
-		// if the column type is a NUMBER, precision is 1, scale is 0, and value is a boolean,
-		// convert it to the integer, when the "true" value is 1, and "false" is 0
-		case oracleColumnData.Type == oracleTypeNumber &&
-			oracleColumnData.Precision != nil && *oracleColumnData.Precision == 1 &&
-			oracleColumnData.Scale != nil && *oracleColumnData.Scale == 0:
-			valueBool, ok := value.(bool)
-			if !ok {
-				result[key] = value
-
-				continue
-			}
-
-			if valueBool {
-				result[key] = 1
-			} else {
-				result[key] = 0
-			}
-
-			continue
-
-		case oracleColumnData.Type == oracleTypeDate:
-			v, err := formatDate(value)
-			if err != nil {
-				return nil, err
-			}
-
-			result[key] = v
-
-			continue
-
-		case strings.Contains(oracleColumnData.Type, oracleTypeTimestamp):
-			v, err := formatTimestamp(value)
-			if err != nil {
-				return nil, err
-			}
+		return QueryPlaceholder, 0, nil
```

Comment:
Wouldn't it make sense to just do `return QueryPlaceholder, valueBool, nil`?

Reply:
No, because the main idea is to convert a boolean type into an integer.
If the `valueBool` is true, I must return an integer `1`.